riscv/build.rs

```diff
@@ -1,12 +1,16 @@
 use std::env;
 
 fn main() {
-    let target = env::var("TARGET").unwrap();
+    println!("cargo::rustc-check-cfg=cfg(riscv)");
+    println!("cargo::rustc-check-cfg=cfg(riscv32)");
```

Comment:
We might want to move these checks (`riscv32` + `riscv64`) into their respective conditional branches.

All the checks pass, so I'm not 100% on how these checks are supposed to work.

Reply:
with `cargo:rustc-check-cfg` you just inform cargo that you might have a custom `cfg` item.